pilot/pkg/networking/core/v1alpha3/configgen.go

```diff
@@ -26,8 +26,9 @@ type ConfigGeneratorImpl struct {
 	Plugins []plugin.Plugin
 	// List of outbound clusters
 	// Must be rebuilt for each push epoch
-	PrecomputedOutboundClusters []*xdsapi.Cluster
-	// TODO: add others in future
+	PrecomputedOutboundClusters  []*xdsapi.Cluster
+	PrecomputedOutboundListeners []*xdsapi.Listener
```

Comment:
How does it interact with the 'isolation' feature ?

Reply:
I need to further improve on this to have a map per config namespace.. See TODO somewhere in the bottom

Reply:
Sounds good.

/lgtm - we can convert to map later, this is already a huge improvement.